app/forms/keywords_form.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_reader :file, :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(csv_file)
+    @file = csv_file
+
+    return false if invalid?
+
+    begin
+      save_keywords_to_db
```

Comment:
How about splitting the method calls? we can use the form attribute inside each method so the methods do not depend on each other and if there's any exception raised inside each method we can easily identify and add a reasonable error to the form

```
    attr_accessor :file, :user, :keywords
```

```suggestion
      parse_keywords // set the keywords here

      save_keywords // use the keywords to save to db here
```

Reply:
Yup I like the idea, keeps the code cleaner and easier to read 😄 Updated here https://github.com/aetheryna/google-scraper-ruby/pull/34/commits/439908c2734f5d8ffecd8a64310ecddee29f9cf6 👷🏾